### DISCUSSION TITLE
UPDATE , on Mac now finally all working now, here the working settings

### DISCUSSION THREAD

**Discussion by Akossimon** [bmaltais/kohya_ss#2570] [Q&A]
UPDATE , on Mac working now

also only now got it running , after 3 weeks of no results. 

this is my environment where it runs now, Apple M2 Pro mit 12‑Core CPU, 19‑Core GPU, 16‑Core Neural Engine and 32GB ram
am doing LORA training of 270 images and it will go for around 60 hours. so 5 days.

i wonder if there are better faster MAc settings in Kohya  that one could do 

below a screenshot pdf (searchable pdf) of all my settings which now worked, so for newbies with these screenshot  it is much easier to take over the settings 

[Kohya SS Mac mimi M2pro 32gb LORA training Settings ABBY OCR.pdf](https://github.com/user-attachments/files/15741535/Kohya.SS.Mac.mimi.M2pro.32gb.LORA.training.Settings.ABBY.OCR.pdf)

[Kohya SS Mac mimi M2pro 32gb LORA training Settings ABBY OCR.pdf]

**Comment by Chadius** [comment]
`MPSGraphUtilities.mm":49:0)): error: 'mps.scatter_nd' op invalid input tensor shape: updates tensor shape and data tensor shape must match along inner dimensions`

Did you ever run into this error? I'm trying to run kohya on my M2 Mac using your settings and this error shows up.

**Reply by Chadius** [reply]
https://github.com/bmaltais/kohya_ss/issues/2185#issuecomment-2030188675

I had to patch some of my files here to get it running.

**Comment by elistys** [comment]
"after 3 weeks of no results."

What were you having trouble with?

I've been trying for 6 hours using the thread below as a reference, but it doesn't work. I haven't tried your settings yet.

https://github.com/bmaltais/kohya_ss/issues/1248

**Comment by Akossimon** [comment]
i have given up

1) it took 36 hours to create only one single Lora,  instead of 10 Loras as set in the prefs , only one, which then did not work.

i use now Draw Things app on Mac from the apple store,  to create loras now, which can be easily made and easily exported and work somewhat ok in auto1111 or in draw things. or anywhere else.

i guess the fact that there is no one helping nor fixing nor improving how to make Koya SS work on a mac,... is the trigger to walk away from this otherwise interesting app. I am sure i am one of many many mac users....who just gave up on Kohya

Strange thing is that on this entire planet only people from the advertising world would have real life application for this Kohya SS , and 100% of all creative directors art directors from such ad agencies only exclusively use mac computers. No one would eve touch a PC.

....and KOHYA ss is exclusively made for PC....so for people who do this for fun, but never ever for any real life applications.

until this KOHYA SS is fixed to work on macs easily and fast, by using its GPU (and so on) correctly , this app will eventually be forgotten and vanish.

anyway...

Thanks for replying....

Akos





On 31 Jul 2024, at 13:30, elistys ***@***.***> wrote:


"after 3 weeks of no results."

What were you having trouble with?

I've been trying for 6 hours using the thread below as a reference, but it doesn't work. I haven't tried your settings yet.

#1248 <https://github.com/bmaltais/kohya_ss/issues/1248>

**Comment by Chadius** [comment]
Yup, I eventually just switched to Runpod. Thanks for looking into this.
